### PR TITLE
Add project tokens for validation and for get requests

### PR DESCRIPTION
**Pull request by Soroosh**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [0.10.1] - 2017-10-10
+### Changed
+- Introduced project tokens
+- allow project tokens to get and push statistiks
```
Comment (Ragnaroek) [inline CHANGELOG.md]: typo statisti_c_s

File: src/fdc_ts/core.clj [hunk 2]
```diff
@@ -51,19 +51,23 @@
 
 ;TODO Move put to separate module
 
-(defn- auth [token ctx]
-  (= (token env) (get-in ctx [:request :headers "auth-token"])))
+(defn- auth [token project-tokens project ctx]
+  (def request-token  (get-in ctx [:request :headers "auth-token"]))
```
Comment (Ragnaroek) [inline src/fdc_ts/core.clj]: I think you are definig request-token globally.
It should be a (let [request-token (get-in ...))
                           (or (= ....)

File: src/fdc_ts/core.clj [hunk 3]
```diff
@@ -51,19 +51,23 @@
...
+      (= (get-in (json/parse-string (project-tokens env))  [project]) request-token)))
 
 (defn- auth-configured [token ctx]
   (token env))
 
-(def auth-publish (partial auth :auth-token-publish))
-(def auth-publish-configured (partial auth-configured :auth-token-publish))
+(def auth-publish (partial auth :auth-token-publish :auth-token-project))
+(def auth-publish-configured (and(partial auth-configured :auth-token-publish)
+                                 (partial auth-configured :auth-token-project)))
```
Comment (Ragnaroek) [inline src/fdc_ts/core.clj]: can you please use a code-formatter like par-infer, par-indent.

File: src/fdc_ts/core.clj [hunk 4]
```diff
@@ -51,19 +51,23 @@
...
 
-(def auth-statistics (partial auth :auth-token-statistics))
-(def auth-statistics-configured (partial auth-configured :auth-token-statistics))
+(def auth-statistics (partial auth :auth-token-statistics :auth-token-project))
+(def auth-statistics-configured (and (partial auth-configured :auth-token-statistics)
```
Comment (Ragnaroek) [inline src/fdc_ts/core.clj]: do not understand the (and ...) here

Comment (Soroosh) [reply src/fdc_ts/core.clj]: Here, we check if the project tokens field is defined in the env. 
auth-statistics-configured is used to see if the service is available.

File: src/fdc_ts/core.clj [hunk 5]
```diff
@@ -59,8 +59,9 @@
 (defn- auth-configured [token ctx]
   (token env))
 
-(def auth-publish
-  (partial auth :auth-token-publish :auth-token-project nil))
+(defn- auth-publish [ctx]
+  (let [project (get-in (:json ctx)  "project")]
+  (auth :auth-token-publish :auth-token-project project ctx)))
```
Comment (Ragnaroek) [inline src/fdc_ts/core.clj]: Hier ist die Einrückung echt verwirrend, das (auth müsste eins eingerück werden, damit man sieht ,dass es zum let-block gehört)

**Pull request merged**